internal/contributebot/webhook/main.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+	"google.golang.org/api/pubsub/v1"
+	"google.golang.org/appengine"
+	"google.golang.org/appengine/log"
+	"google.golang.org/appengine/urlfetch"
+)
+
+func main() {
+	http.HandleFunc("/", handleDefault)
+	http.HandleFunc("/webhook", handleWebhook)
+	appengine.Main()
+}
+
+func handleDefault(w http.ResponseWriter, r *http.Request) {
+	const responseData = `<!DOCTYPE html>
+<title>Go Cloud Contribute Bot</title>
+<h1>Go Cloud Contribute Bot</h1>
+<p>Hello, you've reached <a href="https://github.com/google/go-cloud">Go Cloud</a>'s contribute bot!</p>`
+	if r.URL.Path != "/" {
+		http.NotFound(w, r)
+		return
+	}
+	if r.Method != "GET" && r.Method != "HEAD" {
+		w.Header().Set("Allow", "GET, HEAD")
+		http.Error(w, "Only GET or HEAD allowed", http.StatusMethodNotAllowed)
+		return
+	}
+	w.Header().Set("Content-Length", fmt.Sprint(len(responseData)))
+	w.Header().Set("Content-Type", "text/html; charset=utf-8")
+	if r.Method == "GET" {
+		io.WriteString(w, responseData)
+	}
+}
+
+func handleWebhook(w http.ResponseWriter, r *http.Request) {
```

Comment:
What does this do? Please add a comment to describe this function's behavior.

Reply:
+1 to some comments about what's going on.

Reply:
Done.